### DISCUSSION TITLE
Failed "User-initiated event" WebXR check prevents any further calls to WebXR that would otherwise work

### DISCUSSION THREAD

**Discussion by benferns** [google/model-viewer#3628] [Ideas]
### Description

If activateAR() fails due to WebXR permissions issue (not user-initiated event), future uses of activateAR() will fail to launch WebXR. Ideally the activation would be retried when launched in a different context before falling back, or always retried if "webxr" is the only AR mode set.

There may be nothing that can be done due to the need to fallback or even remove the AR button when webXR is not available, but there are some contexts where what is 'user initiated' can be a bit opaque (e.g. a chains of promises, some reactive frameworks) and multiple pathways to launch webXR might exist with varying behaviour across browsers. 

In fact we discovered this because we were calling 'launch' on page load, which previously worked on certain combinations of Android + Browser versions.

#### Live Demo
https://trapezoidal-tricky-tumbleweed.glitch.me/ (wait for full load)

### Version
- model-viewer: v10.0.1

### Browser Affected

- [x] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [x] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR

- [x] WebXR
- [ ] SceneViewer
- [ ] QuickLook

<img width="655" alt="Screen Shot 2022-03-11 at 14 01 16" src="https://user-images.githubusercontent.com/2927295/157799997-39072025-6f01-4e76-a41e-831a0688d019.png">

**Comment by elalish** [comment]
Hmm, interesting. Do you want to make a PR? I'm not quite sure what behavior would be better. At least this pushed you to find and fix your bug, rather than just having a non-deterministic result, which could have been more easily ignored.
